public/app/features/alerting/unified/GrafanaRuleQueryViewer.tsx

```diff
@@ -98,56 +94,100 @@ export function GrafanaRuleQueryViewer({
 }
 
 interface QueryPreviewProps extends Pick<AlertQuery, 'refId' | 'relativeTimeRange' | 'model'> {
+  rule: CombinedRule;
   isAlertCondition: boolean;
   dataSource?: DataSourceInstanceSettings;
   queryData?: PanelData;
   thresholds?: ThresholdDefinition;
-  evalTimeRange?: RelativeTimeRange;
-  onEvalTimeRangeChange: (timeRange: RelativeTimeRange) => void;
 }
 
 export function QueryPreview({
   refId,
-  relativeTimeRange,
+  rule,
   thresholds,
   model,
   dataSource,
   queryData,
-  evalTimeRange,
-  onEvalTimeRangeChange,
+  relativeTimeRange,
 }: QueryPreviewProps) {
   const styles = useStyles2(getQueryPreviewStyles);
+  const isExpression = isExpressionQuery(model);
+  const [exploreSupported, exploreAllowed] = useAlertRuleAbility(rule, AlertRuleAction.Explore);
+  const canExplore = exploreSupported && exploreAllowed;
+
+  const headerItems: React.ReactNode[] = [];
+
+  if (dataSource) {
+    const dataSourceName = dataSource.name ?? '[[Data source not found]]';
+    const dataSourceImgUrl = dataSource.meta.info.logos.small;
+
+    headerItems.push(<DataSourceBadge name={dataSourceName} imgUrl={dataSourceImgUrl} key="datasource" />);
+  }
 
-  // relativeTimeRange is what is defined for a query
-  // evalTimeRange is temporary value which the user can change
-  const headerItems = [dataSource?.name ?? '[[Data source not found]]'];
   if (relativeTimeRange) {
-    headerItems.push(mapRelativeTimeRangeToOption(relativeTimeRange).display);
+    headerItems.push(
+      <Text color="secondary" key="timerange">
+        {secondsToHms(relativeTimeRange.from)} to now
+      </Text>
+    );
+  }
+
+  let exploreLink: string | undefined = undefined;
+  if (!isExpression && canExplore) {
+    exploreLink = dataSource && createExploreLink(dataSource, model);
   }
 
   return (
-    <QueryBox refId={refId} headerItems={headerItems} className={styles.contentBox}>
-      <pre className={styles.code}>
-        <code>{dump(model)}</code>
-      </pre>
-      {dataSource && (
-        <RuleViewerVisualization
-          refId={refId}
-          dsSettings={dataSource}
-          model={model}
-          data={queryData}
-          thresholds={thresholds}
-          relativeTimeRange={evalTimeRange}
-          onTimeRangeChange={onEvalTimeRangeChange}
-          className={styles.visualization}
-        />
-      )}
-    </QueryBox>
+    <>
+      <QueryBox refId={refId} headerItems={headerItems} exploreLink={exploreLink}>
+        <div className={styles.queryPreviewWrapper}>
+          <ErrorBoundaryAlert>
+            {model && dataSource && <DatasourceModelPreview model={model} dataSource={dataSource} />}
+          </ErrorBoundaryAlert>
+        </div>
+      </QueryBox>
+      {dataSource && <RuleViewerVisualization data={queryData} thresholds={thresholds} />}
+    </>
+  );
+}
+
+function createExploreLink(settings: DataSourceRef, model: AlertDataQuery): string {
```

Comment:
There is an existing `createExploreLink` function https://github.com/grafana/grafana/blob/main/public/app/features/alerting/unified/utils/misc.ts#L34
Could we reuse that or merge somehow?